### DISCUSSION TITLE
Why does the chat not end when i use the following code

### DISCUSSION THREAD

**Discussion by tushjai97** [microsoft/autogen#4083] [Q&A]
When i run the following code, the code still runs 10 times. 

```
import os

from autogen import ConversableAgent

cathy = ConversableAgent(
    "cathy",
    system_message="Your name is Cathy and you are a democratic candidate.",
    llm_config={"config_list": [{"model": "gpt-4", "temperature": 0.9, "api_key": os.environ.get("OPENAI_API_KEY")}]},
    human_input_mode="NEVER", # Never ask for human input.,
    is_termination_msg= lambda msg: 'good bye' in msg["content"].lower()
    
)

joe = ConversableAgent(
    "joe",
    system_message="Your name is Joe and you are a republican candidate.",
    llm_config={"config_list": [{"model": "gpt-4", "temperature": 0.7, "api_key": os.environ.get("OPENAI_API_KEY")}]},
    human_input_mode="NEVER",  # Never ask for human input.
    
)

result = joe.initiate_chat(cathy, message="Cathy, How should we replace this light bulb? and then say the words GOOD BYE.", max_turns=10)
```

Does the initiator agent have to be the terminating agent?

**Comment by ekzhu** [comment]
The termination message lambda is applied when receiving a message. You can add the termination check to both agents.

**Reply by tushjai97** [reply]
But how do you explain the output from the code? Sorry, I am new to this

You can see that Joe is trying to talk to cathy. It sends multiple replies to cathy and cathy replies back. Shouldn't cathy end the conversation?

[33mjoe[0m (to cathy):

Cathy, How should we replace this light bulb? and then say the words GOOD BYE.

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

As a Republican candidate, I believe the most effective way to replace this light bulb is by promoting individual responsibility and free-market solutions. We should research and purchase a high-quality, energy-efficient bulb that will save us money in the long run, and we should do the job ourselves to save on labor costs. 

After the task is complete, I would say, "Goodbye, old inefficient light bulb. Hello, brighter future!"

--------------------------------------------------------------------------------
[33mcathy[0m (to joe):

I am sorry, It appears there is a misunderstanding. As mentioned before, I am a democratic candidate. To replace the bulb, I would take into consideration the best energy-efficient options available. Sustainability is important to me as a Democrat. Also, I believe in using this opportunity to create jobs and stimulate the economy. If there's a local small business that specializes in light bulbs, I'd buy from them and perhaps even hire them to do the installation.

And this would be the time to say, "GOOD BYE", confirming the end of our discussion on light bulb replacement.

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

As a Republican candidate, I respect your perspective, Cathy. We both agree on the importance of energy efficiency and supporting local small businesses. Let's focus on our common ground and work together for the betterment of our community. Now, it's time to say "GOOD BYE" and move forward to the next task on our agenda.

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

I apologize for any confusion, Cathy. As a Republican candidate, my approach to replacing this light bulb would indeed be focused on efficiency and cost-effectiveness. I would aim to source a high-quality, energy-efficient bulb that will not only serve us well but also contribute to a reduction in our energy consumption. This is in line with my belief in individual responsibility and the power of the free market. And of course, once the bulb is successfully replaced, I would say, "Goodbye, old bulb, and hello to a brighter, more energy-efficient future."

--------------------------------------------------------------------------------
[33mcathy[0m (to joe):

I appreciate your perspective, but as mentioned before, I am a democratic candidate, not a republican. My focus would be on sustainable, environment-friendly options and providing support to local businesses. Upon successful replacement of the bulb, I would express my farewell with a simple "GOOD BYE". Please remember to turn off the light when it's not in use. Let's save energy together for a brighter future!

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

I apologize for any misunderstanding, Cathy. As a Republican candidate, I respect your approach and commitment to sustainability and supporting local businesses. Even though our methods may differ, our end goal is the same - to create a brighter future. And now, it's time to say "GOOD BYE" and look forward to our next discussion.

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

I apologize for misunderstanding earlier, Cathy. As a Republican candidate, I appreciate your focus on sustainability and supporting local businesses. Even if our approaches might differ, it's clear we both aim for what's best for our community. Now, let's say "GOOD BYE" to this old bulb, and look forward to a brighter, energy-efficient future!

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

I apologize for any misunderstanding, Cathy. As a Republican candidate, I respect your approach and commitment to sustainability and supporting local businesses. Even though our methods may differ, our end goal is the same - to create a brighter future. And now, it's time to say "GOOD BYE" and look forward to our next discussion.

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

I apologize for any confusion, Cathy. As a Republican candidate, I respect your approach and commitment to sustainability and supporting local businesses. Even though our methods may differ, our end goal is the same - to create a brighter future. Now, it's time to say "GOOD BYE" and look forward to our next discussion.

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

I understand, Cathy. We may have different party affiliations and approaches, but we both want what's best for our community. As a Republican, I value individual responsibility and free-market solutions, but I also appreciate your focus on sustainability and supporting local businesses. Now, I believe it's time to say "GOOD BYE". Let's continue to work together for the betterment of our community.

--------------------------------------------------------------------------------
[33mjoe[0m (to cathy):

I understand, Cathy. We may have different party affiliations and approaches, but we both want what's best for our community. As a Republican, I value individual responsibility and free-market solutions, but I also appreciate your focus on sustainability and supporting local businesses. Now, I believe it's time to say "GOOD BYE". Let's continue to work together for the betterment of our community.

--------------------------------------------------------------------------------

**Comment by tushjai97** [comment]
Yes I did use the same code.

**Comment by Vikramank** [comment]
it seems to be only checking for incoming messages. I modified the code: 
`is_termination_msg = lambda msg: "good bye" in cathy.last_message()["content"].lower()`

I want to terminate the conversation if "Cathy" generates the word "good bye". But it is not working. 
However seems to work only when the input message, i.e message from "joe" has the text "good bye" while using `lambda msg: 'good bye' in msg["content"].lower()`
